Helper/MetricsClient.php

```diff
@@ -321,9 +331,19 @@ public function postMetrics()
         if (!$this->configHelper->shouldCaptureMetrics()) {
             return null;
         }
+        $previousPostTime = $this->loadFromCache(self::METRICS_TIMESTAMP_ID);
+        // checks if key exists
+        if (!$previousPostTime) {
+            $this->saveToCache(self::METRICS_TIMESTAMP_ID, round(microtime(true) * 1000));
+        } else {
+            $timeDiff = round(microtime(true) * 1000) - $previousPostTime;
+            // if its under 30 seconds since last key don't do anything
+            if ($timeDiff < self::METRICS_POST_INTERVAL) {
+                return null;
+            }
+        }
         $workingFile = null;
```

Comment:
just for my understanding, if two threads call `postMetrics` around the same time, it's possible that two threads reach here (because `METRICS_TIMESTAMP_ID` cache is only updated after successful post). In such case, what would happen?